crates/hyperswitch_connectors/src/connectors/fiservemea/transformers.rs

```diff
@@ -56,66 +92,261 @@ impl TryFrom<&FiservemeaRouterData<&PaymentsAuthorizeRouterData>> for Fiservemea
+    response_code: Option<String>,
+    response_message: Option<String>,
+    avs_response: Option<AvsResponse>,
+    security_code_response: Option<String>,
+}
+
+fn map_status(
+    fiservemea_status: Option<FiservemeaPaymentStatus>,
+    fiservemea_result: Option<FiservemeaPaymentResult>,
+    transaction_type: FiservemeaTransactionType,
+) -> Result<common_enums::AttemptStatus, errors::ConnectorError> {
+    match fiservemea_status {
+        Some(status) => match status {
+            FiservemeaPaymentStatus::Approved => {
+                if transaction_type == FiservemeaTransactionType::Preauth {
+                    Ok(common_enums::AttemptStatus::Authorized)
+                } else if transaction_type == FiservemeaTransactionType::Void {
+                    Ok(common_enums::AttemptStatus::Voided)
+                } else if matches!(
+                    transaction_type,
+                    FiservemeaTransactionType::Sale | FiservemeaTransactionType::Postauth
+                ) {
+                    Ok(common_enums::AttemptStatus::Charged)
+                } else {
+                    Ok(common_enums::AttemptStatus::Failure)
+                }
+            }
+            FiservemeaPaymentStatus::Waiting => Ok(common_enums::AttemptStatus::Pending),
+            FiservemeaPaymentStatus::Partial => Ok(common_enums::AttemptStatus::PartialCharged),
+            FiservemeaPaymentStatus::ValidationFailed
+            | FiservemeaPaymentStatus::ProcessingFailed
+            | FiservemeaPaymentStatus::Declined => Ok(common_enums::AttemptStatus::Failure),
+        },
+        None => match fiservemea_result {
+            Some(result) => match result {
+                FiservemeaPaymentResult::Approved => {
+                    if transaction_type == FiservemeaTransactionType::Preauth {
+                        Ok(common_enums::AttemptStatus::Authorized)
+                    } else if transaction_type == FiservemeaTransactionType::Void {
+                        Ok(common_enums::AttemptStatus::Voided)
+                    } else if matches!(
+                        transaction_type,
+                        FiservemeaTransactionType::Sale | FiservemeaTransactionType::Postauth
+                    ) {
+                        Ok(common_enums::AttemptStatus::Charged)
+                    } else {
+                        Ok(common_enums::AttemptStatus::Failure)
+                    }
+                }
+                FiservemeaPaymentResult::Waiting => Ok(common_enums::AttemptStatus::Pending),
+                FiservemeaPaymentResult::Partial => Ok(common_enums::AttemptStatus::PartialCharged),
+                FiservemeaPaymentResult::Declined
+                | FiservemeaPaymentResult::Failed
+                | FiservemeaPaymentResult::Fraud => Ok(common_enums::AttemptStatus::Failure),
+            },
+            None => Err(errors::ConnectorError::MissingRequiredField {
+                field_name: "transactionResult",
+            }),
```

Comment:
This will trigger 4xx and mark payment as failure, please check and remove this error